src/components/genericTable/genericTable.tsx

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { useState } from 'react';
+import Styles from './genericTable.module.scss';
+import { Box, Paper, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, TableSortLabel } from '@mui/material';
+
+export type TableColumn<T> = {
+  id: keyof T | 'actions';
+  label: string;
+  sortable?: boolean;
+  minWidth?: number;
+  align?: 'right' | 'left' | 'center';
+  format?: (value: any) => string;
```

Comment:
as we talked, only use eslint-disable on this.

Reply:
found the right generic type :
`export type TableColumn<T extends PrimitiveObject, K extends keyof T = keyof T> = {
  id: K | 'actions';
  label: string;
  sortable?: boolean;
  minWidth?: number;
  align?: 'right' | 'left' | 'center';
  format?: (value: T[K]) => string;
  render?: (raw: T) => React.ReactNode;
};`